packages/@react-stately/tooltip/src/TooltipManager.ts

```diff
@@ -10,67 +10,76 @@
  * governing permissions and limitations under the License.
  */
 
-const  TOOLTIP_DELAY = 220; // --spectrum-global-animation-duration-400
+import {MultipleTolltipManager, TooltipState} from './types';
 
-export class TooltipManager {
+const CLOSE_DELAY = 220; // --spectrum-global-animation-duration-400
+const WARMUP_DELAY = 350;  // --spectrum-global-animation-duration-700
+const COOLDOWN_DELAY = 400; // --spectrum-global-animation-duration-1000
 
-  visibleTooltip?: null | {triggerId: string, state: {open: boolean, setOpen(value: boolean), tooltipManager: string}}
-  hoverHideTimeout?: null | ReturnType<typeof setTimeout>;
-  hoverShowTimeout?: null | ReturnType<typeof setTimeout>;
+let visibleTooltip: null | TooltipState = null;
+let warmupTimer: null | ReturnType<typeof setTimeout> = null;
+let cooldownTimer: null | ReturnType<typeof setTimeout> = null;
+let closeTimer: null | ReturnType<typeof setTimeout> = null;
+let warmedup: boolean = false;
 
-  // Arbitrary timeout lengths in place for current demo purposes. Delays to be adjusted for warmup / cooldown logic PR
-  constructor() {
-    this.visibleTooltip = null;
-    this.hoverHideTimeout = null;
-    this.hoverShowTimeout = null;
-  }
+export class TooltipManager implements MultipleTolltipManager {
+  private state: TooltipState;
 
-  updateTooltipState(state, triggerId)  {
-    state.setOpen(!state.open);
-    this.visibleTooltip = {triggerId, state};
+  constructor(state: TooltipState) {
+    this.state = state;
   }
 
-  isSameTarget(currentTriggerId) {
-    return currentTriggerId === this.visibleTooltip.triggerId;
+  get isOpen(): boolean {
+    return this.state.isOpen;
   }
 
-  showTooltip(state) {
-    state.setOpen(true);
+  openTooltip() {
     // Close previously open tooltip
-    if (this.visibleTooltip) {
-      this.visibleTooltip.state.setOpen(false);
+    if (visibleTooltip) {
+      visibleTooltip.close();
     }
+
+    this.state.open();
+    visibleTooltip = this.state;
```

Comment:
Noticed that it is possible to have multiple tooltips open at once through the following flow:
1. Hover over the copy button to trigger a tooltip opening
2. Press "Tab" to focus another button. Note that the previous tooltip closes and the newly focused button shows its tooltip (this is correct behavior so far).
3. Hover over a different button. Note that two tooltips are now showing

I'll need to dig into why this is happening, but noting it here for now

Reply:
good catch...  I looked into, but didnt find real reason why it happens.... maybe to open separate ticket for this one